chia/util/cached_bls.py

```diff
@@ -0,0 +1,49 @@
+import functools
+from typing import List, Optional
+
+from blspy import AugSchemeMPL, G1Element, G2Element, GTElement
+from chia.util.hash import std_hash
+from chia.util.lru_cache import LRUCache
+
+
+def get_pairings(cache: LRUCache, pks: List[G1Element], msgs: List[bytes], force: bool) -> List[GTElement]:
```

Comment:
`force` is a bit ambiguous about what it's forcing. It seems to be forcing the values to go into cache even if the heuristic suggests they should? Maybe `force_despite_heuristic` or `always_cache` or something?

Reply:
I wonder if you could just skip the heuristic altogether. The three cases I can think of are unfinished block, finished block, and mempool. It seems to me we could get all the benefit if we always write to cache except in the finished block case, where we never write new values to cache since we never expect to see them again unless we reorg. Am I missing something? @mariano54

Reply:
Renamed to `force_cache`.

Reply:
The finished block case is two different cases (syncing and normal operation),
In the normal operation case, we need to USE the cache, but don't necesarily need to store anything in it.
In syncing case, we don't even want to use the cache, since that will slow down validation.